server/players.lua

```diff
@@ -76,9 +87,18 @@ RegisterNetEvent('ps-adminmenu:server:SetGang', function(data, selectedData)
     local playerId, Gang, Grade = selectedData["Player"].value, selectedData["Gang"].value, selectedData["Grade"].value
     local Player = QBCore.Functions.GetPlayer(playerId)
     local name = Player.PlayerData.charinfo.firstname..' '..Player.PlayerData.charinfo.lastname
+    local GangInfo = QBCore.Shared.Gangs[Gang]
+    if GangInfo then
+        if GangInfo["grades"][selectedData["Grade"].value] then
+            Player.Functions.SetGang(tostring(Gang), tonumber(Grade))
+            QBCore.Functions.Notify(src, locale("gangset", name, Gang, Grade), 'success', 5000)
+        else
+            TriggerClientEvent('QBCore:Notify', source, "Not a valid grade", 'error')
+        end
+    else
+        TriggerClientEvent('QBCore:Notify', source, "Not a valid Gang", 'error')
+    end
```

Comment:
This can get the same guard clauses "treatment" as above. Let me know if you have questions.